experimental/devices/mfrc522/mfrc522.go

```diff
@@ -514,156 +684,19 @@ func (r *Dev) SelectTag(serial []byte) (byte, error) {
+func (r *Dev) stopCrypto() error {
 	return r.clearBitmask(commands.Status2Reg, 0x08)
 }
 
-// ReadBlock reads the block from the card.
+// readBlock reads the block from the card.
 //
 // 	sector - card sector to read from
 // 	block - the block within the sector (0-3 tor Mifare 4K)
-func (r *Dev) ReadBlock(sector int, block int) ([]byte, error) {
+func (r *Dev) readBlock(sector int, block int) ([]byte, error) {
 	return r.read(calcBlockAddress(sector, block%3))
 }
 
-// WriteBlock writes the data into the card block.
-//
-// 	auth - the authentiction mode.
-// 	sector - the sector on the card to write to.
-// 	block - the block within the sector to write into.
-// 	data - 16 bytes if data to write
-// 	key - the key used to authenticate the card - depends on the used auth method.
-func (r *Dev) WriteBlock(auth byte, sector int, block int, data [16]byte, key [6]byte) (err error) {
-	defer func() {
-		if err == nil {
-			err = r.StopCrypto()
-		}
-	}()
-	uuid, err := r.selectCard()
-	if err != nil {
-		return
-	}
-	state, err := r.Auth(auth, sector, 3, key, uuid)
-	if err != nil {
-		return
-	}
-	if state != AuthOk {
-		err = wrapf("authentication failed")
-		return
-	}
-
-	return r.write(calcBlockAddress(sector, block%3), data[:])
-}
-
-// ReadSectorTrail reads the sector trail (the last sector that contains the
-// sector access bits)
-//
-// 	sector - the sector number to read the data from.
-func (r *Dev) ReadSectorTrail(sector int) ([]byte, error) {
-	return r.read(calcBlockAddress(sector&0xFF, 3))
-}
-
-// WriteSectorTrail writes the sector trait with sector access bits.
-//
-// 	auth - authentication mode.
-// 	sector - sector to set authentication.
-// 	keyA - the key used for AuthA authentication scheme.
-// 	keyB - the key used for AuthB authentication schemd.
-// 	access - the block access structure.
-// 	key - the current key used to authenticate the provided sector.
-func (r *Dev) WriteSectorTrail(auth byte, sector int, keyA [6]byte, keyB [6]byte, access *BlocksAccess, key [6]byte) (err error) {
-	defer func() {
-		if err == nil {
-			err = r.StopCrypto()
-		}
-	}()
-	uuid, err := r.selectCard()
-	if err != nil {
-		return
-	}
-	state, err := r.Auth(auth, sector, 3, key, uuid)
-	if err != nil {
-		return
-	}
-	if state != AuthOk {
-		err = wrapf("failed to authenticate")
-		return
-	}
-
-	var data [16]byte
-	copy(data[:], keyA[:])
-	accessData := CalculateBlockAccess(access)
-	copy(data[6:], accessData[:4])
-	copy(data[10:], keyB[:])
-	return r.write(calcBlockAddress(sector&0xFF, 3), data[:])
-}
-
-// Auth authenticate the card fof the sector/block using the provided data.
-//
-// 	mode - the authentication mode.
-// 	sector - the sector to authenticate on.
-// 	block - the block within sector to authenticate.
-// 	sectorKey - the key to be used for accessing the sector data.
-// 	serial - the serial of the card.
-func (r *Dev) Auth(mode byte, sector, block int, sectorKey [6]byte, serial []byte) (AuthStatus, error) {
```

Comment:
This function may is required to allow testing the authentication for the sector without reading it, would be nice to keep it,